### ISSUE TITLE
Implement ConvTranspose2D for TTNN (needed for vanilla UNet variants)

### ISSUE THREAD

**Issue by dvartaniansTT** [tenstorrent-metal/tt-metal#6326]
**Is your feature request related to a problem? Please describe.**
Given the shallow UNet does not have actual trained weights, in order to demo our efforts for the UNet model, we need to bring up the vanilla variant/s [Brain MRI segmentation](https://pytorch.org/hub/mateuszbuda_brain-segmentation-pytorch_unet/) or the Carvana variant on device. 
both variants have an additional op [convTranspose2D pytorch op](https://pytorch.org/docs/stable/generated/torch.nn.ConvTranspose2d.html). 
In the shallow variant we have the upsample2D instead. Given convTranspose2D has learned params, we need to have the equivalent op in TTNN to run the models end to end on device. 

**Describe the solution you'd like**
Have the [convTranspose2d op](https://pytorch.org/docs/stable/generated/torch.nn.ConvTranspose2d.html)  kernel implementation and make it callable in TTNN. 

**Note**
currently developing the demo on GS - e150.

**Comment by jliangTT** [comment]
Assigning to @nsmithtt to triage - putting this offline while we discuss the priority of these items offline.

**Comment by nsmithtt** [comment]
@davorchap, it's not that simple, a transpose conv stride 2 corresponds to dilating the activations by a factor of 2, this is like upsample but with 0 rows and columns interleaved in the activation.  I agree that halo infra is the right place for it, but we don't have a mechanism that does this, it's probably a week's worth of work because of corner cases and debug.

**Comment by dvartaniansTT** [comment]
Now, this is required for customer models as well

**Comment by dvartaniansTT** [comment]
@keerthana-r-mcw and @saichandax please share the convTranspose2D torch ops that we need for modelNet in this ticket.

**Comment by HariniMohan0102** [comment]
@dvartaniansTT Here are the configurations of convTranspose2D of model_net:
**_Note_**: The following ops are tensorflow op details, the input and output shapes are in NHWC format. 

High resolution (4094x510):
<google-sheets-html-origin>
Input Shape | Output Shape | Kernel Size | Stride | Padding | Dilation | Groups | Activation
-- | -- | -- | -- | -- | -- | -- | --
(None, 254, 30, 64) | (None, 510, 62, 64) | (4, 4) | (2, 2) | valid | (1, 1) | 1 | relu
(None, 506, 58, 128) | (None, 1014, 118, 64) | (4, 4) | (2, 2) | valid | (1, 1) | 1 | relu
(None, 1010, 114, 128) | (None, 2022, 230, 64) | (4, 4) | (2, 2) | valid | (1, 1) | 1 | relu
(None, 2018, 226, 128) | (None, 4038, 454, 2) | (4, 4) | (2, 2) | valid | (1, 1) | 1 | relu

Low Resolution (2047x255):
<google-sheets-html-origin>
Input Shape | Output Shape | Kernel Size | Stride | Padding | Dilation | Groups | Activation
-- | -- | -- | -- | -- | -- | -- | --
(None, 126, 14, 64) | (None, 254, 30, 64) | (4, 4) | (2, 2) | valid | (1, 1) | 1 | relu
(None, 250, 26, 128) | (None, 502, 54, 64) | (4, 4) | (2, 2) | valid | (1, 1) | 1 | relu
(None, 498, 50, 128) | (None, 998, 102, 64) | (4, 4) | (2, 2) | valid | (1, 1) | 1 | relu
(None, 994, 98, 128) | (None, 1990, 198, 2) | (4, 4) | (2, 2) | valid | (1, 1) | 1 | relu

**Comment by punithsekar** [comment]
I am currently looking at the vanilla UNet model.
To track the issue I am tagging myself: @punithsekar

**Comment by punithsekar** [comment]
@sankarmanoj-tt, I see the ConvTranspose2D PR #13639 has been merged to main. I have tested the operation with Vanilla Unet configurations. Out of four test cases, one has been failing due to OOM, for which I have created a separate issue #14882.

Can We Close this ticket and track it there?

CC: @dvartaniansTT @saichandax
